observations/obs_converters/Ameriflux/fluxnetfull_to_obs.f90

```diff
@@ -0,0 +1,1180 @@
+tower%hQC         =      nint(values(tower%hQCindex        -time_adjust))
+tower%gppDT       =           values(tower%gppDTindex      -time_adjust)
+tower%gppNT       =           values(tower%gppNTindex      -time_adjust)
+tower%recoDT      =           values(tower%recoDTindex     -time_adjust)
+tower%recoNT      =           values(tower%recoNTindex     -time_adjust)
+tower%gppDTUNC16  =           values(tower%gppDTUNC16index -time_adjust)
+tower%gppDTUNC84  =           values(tower%gppDTUNC84index -time_adjust)
+tower%gppNTUNC16  =           values(tower%gppNTUNC16index -time_adjust)
+tower%gppNTUNC84  =           values(tower%gppNTUNC84index -time_adjust)
+tower%recoDTUNC16 =           values(tower%recoDTUNC16index-time_adjust)
+tower%recoDTUNC84 =           values(tower%recoDTUNC84index-time_adjust)
+tower%recoNTUNC16 =           values(tower%recoNTUNC16index-time_adjust)
+tower%recoNTUNC84 =           values(tower%recoNTUNC84index-time_adjust)
+deallocate(values)
+
+
+
+read(tower%start_time(1:12), fmt='(i4, 4i2)') yeara,montha,daya,houra,mina 
+read(tower%end_time(1:12),   fmt='(i4, 4i2)') yearb,monthb,dayb,hourb,minb
+
+
+! Certain time resolutions (MM) does not define days
+if (time_resolution == 'MM') then
+   daya = 1
+   dayb = 1
+endif
+
+
+
+write(*,*)''
+write(string1, *) 'Display tower%start_time,yeara,montha,daya,houra,mina (LTC)  =', tower%start_time, yeara, montha, daya, houra, mina
+write(string2, *) 'Display tower%end_time,yearb,monthb,dayb,hourb,minb   (LTC)  =', tower%end_time, yearb, monthb, dayb, hourb, minb
+if (verbose) call error_handler(E_MSG,'Fluxnetfull_to_obs',string1,text2=string2)
+
+
+
+date_start= set_date(yeara,montha,daya,houra,mina,0)
+date_end=   set_date(yearb,monthb,dayb,hourb,minb,0)
+
+
+
+! Assign average of flux time window to obs_seq (DART time)
+tower%time_obs = (date_start+date_end) / 2
+
+! Covert from Fluxnet provided LTC to UTC, UTC is standard for DART and CLM 
+! For example, EST = UTC-5 
+if (timezoneoffset < 0.0_r8) then
+   tower%time_obs = tower%time_obs + offset
+else
+   tower%time_obs = tower%time_obs - offset
+endif
+
+
+! Reject NEE data where neeQC is missing
+if (tower%neeQC < 0) tower%neeQC = maxgoodqc + 100
+
+! If NEE (DD,WW,MM) convert from  % filled QC to integer QC  good/fair/poor
+if (time_resolution == 'DD' .or. time_resolution == 'MM' .or. &
+    time_resolution == 'WW') then
+   if (tower%neeQCfrac >= 0.90) tower%neeQC = 1 ! >90% fill is good
+   if (tower%neeQCfrac < 0.90 .and. tower%neeQC >= 0.60) tower%neeQC = 2 ! 60-90% fill is fair
+   if (tower%neeQCfrac < 0.60 .and. tower%neeQC >= 0.0_r8) tower%neeQC = 3 ! <60% fill is poor
+endif
+
+
+! The QC values are typically missing for le and h (-9999)
+! Thus  manually assign poor QC values in these cases
+if (energy_balance .eqv. .false.) then
+   if (tower%leQC  < 0.0_r8) tower%leQC  = 2
+   if (tower%hQC   < 0.0_r8) tower%hQC   = 2
+else  ! No QC values for energy balance corrected le and h.  Assign fair QC.
```

Comment:
I assume you are trying to say
if (tower%leQC < 0.0_r8) tower%leQC = 2
if(tower%hQC   <0.0_r8)  tower%hQC = 2

Reply:
When energy_balance is 'off' the leQC and hQC values can be missing (-9999), but not always, so in some instances the code has to make a decision what to do in case of missing QC and the observation still looks OK.  My subjective decision was to assign a QC=2 which puts the decision back to the user during the assimilation step of what to do with that data.

When energy_balance is 'on' there are no QC values at all for the 'LE_CORR' and H_CORR' variables, so the code has to always make a decision of what the QC is, which I subjectively assign as 2.

Reply:
Okay, I see. I didn't know when the energy_balance is 'on' there are no QC values for 'LE_CORR' and H_CORR'.